pkg/integrations/prometheus/promql/metricsSearchHandler.go

```diff
@@ -147,52 +149,124 @@ func parseSearchBody(jsonSource map[string]interface{}) (string, uint32, uint32,
+	if timeParam == "" {
+		log.Errorf("parseTimeForPromQL: time parameter is empty")
+		return 0, errors.New("time parameter is empty")
+	}
+	var timeValue int64
+
+	if timeParam != "" {
+		if parsedInt, err := strconv.ParseInt(timeParam, 10, 64); err == nil {
+			// If timeParam can be parsed as an integer, use it as the time value
+			timeValue = parsedInt
+
+		} else if parsedTime, err := time.Parse(time.RFC3339, timeParam); err == nil {
+			// If timeParam can be parsed as an RFC3339 time, use it as the time value
+			timeValue = parsedTime.Unix()
+		} else {
+			return 0, err
+		}
 	}
+
+	return uint32(timeValue), nil
+}
+func ProcessPromqlMetricsSearchRequest(ctx *fasthttp.RequestCtx, myid uint64) {
 	qid := rutils.GetNextQid()
+	searchText := string(ctx.FormValue("query"))
+	timeParam := string(ctx.FormValue("time"))
 
-	readJSON := make(map[string]interface{})
-	var jsonc = jsoniter.ConfigCompatibleWithStandardLibrary
-	decoder := jsonc.NewDecoder(bytes.NewReader(rawJSON))
-	decoder.UseNumber()
-	err := decoder.Decode(&readJSON)
+	var endTime uint32
+	var err error
+
+	if searchText == "" {
+		log.Errorf("ProcessMetricsSearchRequest: no query parameter provided")
+		utils.SetBadMsg(ctx, "query is required")
+		return
+	}
+
+	if timeParam == "" {
+		// If timeParam doesn't exist, assume the current time in epoch seconds as endTime
+		endTime = uint32(time.Now().Unix())
+	} else {
+		endTime, err = parseTimeForPromQL(timeParam)
+		if err != nil {
+			log.Errorf("ProcessPromqlMetricsSearchRequest: Error parsing time parameter, err:%v", err)
+			return
+		}
+	}
+	log.Infof("qid=%v, ProcessMetricsSearchRequest:  searchString=[%v] startEpochs=[%v] endEpochs=[%v]", qid, searchText, endTime-1, endTime)
+
+	metricQueryRequest, pqlQuerytype, _, err := convertPqlToMetricsQuery(searchText, endTime-108000, endTime, myid)
 	if err != nil {
 		ctx.SetContentType(ContentJson)
 		ctx.SetStatusCode(fasthttp.StatusBadRequest)
 		WriteJsonResponse(ctx, nil)
+		log.Errorf("qid=%v, ProcessMetricsSearchRequest: Error parsing query err=%+v", qid, err)
 		_, err = ctx.WriteString(err.Error())
 		if err != nil {
 			log.Errorf("qid=%v, ProcessMetricsSearchRequest: could not write error message err=%v", qid, err)
 		}
-		log.Errorf("qid=%v, ProcessMetricsSearchRequest: failed to decode search request body! Err=%+v", qid, err)
 		return
 	}
+	segment.LogMetricsQuery("PromQL metrics query parser", &metricQueryRequest[0], qid)
+	res := segment.ExecuteMetricsQuery(&metricQueryRequest[0].MetricsQuery, &metricQueryRequest[0].TimeRange, qid)
 
-	searchText, startTime, endTime, step, _, err := parseSearchBody(readJSON)
+	mQResponse, err := res.GetResultsPromQl(&metricQueryRequest[0].MetricsQuery, pqlQuerytype)
 	if err != nil {
-		ctx.SetContentType(ContentJson)
-		ctx.SetStatusCode(fasthttp.StatusBadRequest)
-		_, err = ctx.WriteString(err.Error())
+		log.Errorf("ExecuteAsyncQuery: Error getting results! %+v", err)
```

Comment:
nit: funcname is not correct, it sould be `ProcessPromqlMetricsSearchRequest: .. failed to executeAsyncQuery...`